proxy/src/main/java/com/wavefront/agent/QueuedAgentService.java

```diff
@@ -430,10 +431,18 @@ private void addTaskToSmallestQueue(ResubmissionTask taskToRetry) {
     TaskQueue<ResubmissionTask> queue = getSmallestQueue();
     if (queue != null) {
       synchronized (queue) {
-        queue.add(taskToRetry);
+        try {
+          queue.add(taskToRetry);
+        } catch (FileException e) {
+          logger.warning(
+              "CRITICAL (Losing points!): " +
+                  "WF-1: Could not open enough file descriptors to send points to " +
```

Comment:
This is not the right message, this is actually it running out of queue space and not FDs. We have another way to detect this by looking at the actual FD usage so I think we should just say that we can't write to the queue any more (due to space) and we should definitely add the exception to the message as well